### ISSUE TITLE
Parse error in multiline dict with equal sign

### ISSUE THREAD

**Issue by Scony** [godotengine/godot#35416]
**Godot version:**
master @ d11d7dfe3e15e34ba400e6b73d618abc4b8f5886

**OS/device including version:**
Linux

**Issue description:**
Godot raises parse error on the code below:
```
var y = { name = 1 }			# fine
var x = {
	name = # parse errror
	1
}
```
```
Godot Engine v3.2.rc.custom_build.d11d7dfe3 - https://godotengine.org
 
SCRIPT ERROR: GDScript::reload: Parse Error: Error parsing expression, misplaced: '\n'
   At: res://tests/potential-godot-bugs/multiline-dict-w-eq-sign.gd:4.
ERROR: reload: Method/Function Failed, returning: ERR_PARSE_ERROR
   At: modules/gdscript/gdscript.cpp:576.
```

**Comment by Calinou** [comment]
I think this is expected. If you want to do this, you have to add a backslash at the end of the line or wrap the expression between parentheses.

**Comment by Scony** [comment]
To me it's weird especially because
```
var x = {
    'name':
    1
}
``` 
is working fine. Naturally, as a consequence, the same rule should apply to `=`.

**Comment by bojidar-bg** [comment]
The issue comes from a fallthrough in the dictionary parsing, which short-circuits from key to value, and fails to consume the newline at [`gdscript_parser.cpp:1096`](https://github.com/godotengine/godot/blob/1c6d1fcf956b882fc8a14fa595b7f1e9b15e1574/modules/gdscript/gdscript_parser.cpp#L1095-L1097).
(newline is consumed at [`gdscript_parser.cpp:1028`](https://github.com/godotengine/godot/blob/1c6d1fcf956b882fc8a14fa595b7f1e9b15e1574/modules/gdscript/gdscript_parser.cpp#L1028-L1030) after a colon, note that `DICT_EXPECT_KEY/VALUE` are within an `else`).

The best solution, apart from rewriting the dictionary parsing state machine, would be to increment/decrement `parenthesis` while inside the dictionary.

**Comment by KoBeWi** [comment]
Fixed in 4.0
